pkgs/development/python-modules/xdot/default.nix

```diff
@@ -10,29 +10,34 @@
   graphviz,
   gtk3,
   numpy,
+  packaging,
+  python3Packages,
 }:
 
 buildPythonPackage rec {
   pname = "xdot";
-  version = "1.3";
-  format = "setuptools";
+  version = "1.4";
+  pyproject = true;
 
   src = fetchFromGitHub {
     owner = "jrfonseca";
     repo = "xdot.py";
     rev = version;
-    hash = "sha256-0UfvN7z7ThlFu825h03Z5Wur9zbiUpvD5cb5gcIhQQI=";
+    hash = "sha256-fkO1bINRkCCzVRrQg9+vIODbN+bpXq2OHBKkzzZUZNA=";
   };
 
+  build-system = [ python3Packages.setuptools ];
+
   nativeBuildInputs = [
     gobject-introspection
     wrapGAppsHook3
-  ];
-  propagatedBuildInputs = [
-    pygobject3
     graphviz
     gtk3
```

Comment:
@widlarizer indeed you almost resolved @dotlambda 's request to not propagate these, but these also shouldn't be in `nativeBuildInputs`, but rather in `buildInputs`. Could you please do that? Other then that it looks good to me.

Reply:
How can I observe this to be incorrect? Would this only prove problematic on cross compilation? Should I test packages by cross compiling them?

Reply:
> How can I observe this to be incorrect?

Usually this is a matter of common sense. You gain this experience by knowing the role of the different dependencies after a while contributing Nixpkgs.

> Would this only prove problematic on cross compilation?

Yes, probably.

> Should I test packages by cross compiling them?

Not necessarily, as good reviewers should spot these issues. However if you are not sure, for simple packages like this it can be a good idea.

You can also try to figure this out yourself by reading the docs:

https://nixos.org/manual/nixpkgs/unstable/#ssec-stdenv-dependencies-overview